conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/bitcoin/BitcoinOperations.scala

```diff
@@ -0,0 +1,100 @@
+package tech.cryptonomic.conseil.indexer.bitcoin
+
+import cats.effect.{Concurrent, Resource}
+import com.typesafe.scalalogging.LazyLogging
+import fs2.Stream
+import slick.jdbc.PostgresProfile.api._
+import slickeffect.Transactor
+
+import tech.cryptonomic.conseil.common.config.Platforms.BitcoinBatchFetchConfiguration
+import tech.cryptonomic.conseil.common.rpc.RpcClient
+import tech.cryptonomic.conseil.common.bitcoin.{BitcoinPersistence, Tables}
+import tech.cryptonomic.conseil.common.bitcoin.rpc.BitcoinClient
+import tech.cryptonomic.conseil.indexer.config.{Custom, Depth, Everything, Newest}
+
+/**
+  * Bitcoin operations for Lorre.
+  *
+  * @param bitcoinClient Bitcoin client instance
+  * @param persistence Bitcoin persistence instance
+  * @param tx [[slickeffect.Transactor]] to perform a Slick operations on the database
+  */
+class BitcoinOperations[F[_]: Concurrent](
```

Comment:
@ivanopagano what would be an advantage of that?

Reply:
What's the advantage of doing it generally, to be honest?

I've trodden this road before and I have the feeling that the constraint get so strict at a certain point, that keeping the abstraction makes sense only for the basic building blocks, but not at the top.
After all, once you get Concurrent/Effect/ConcurrentEffect, how would you think to implement that?

I mean, are you thinking of something different from IO, eventually?

Reply:
I just asked because for me it makes no difference. It's like comparing colours of the same model of a car - engine is the same, but some people like it blue, and some people like it red.